test/extended/operators/server_side_apply_examples.go

```diff
@@ -0,0 +1,93 @@
+package operators
+
+import (
+	"context"
+
+	applyconfigv1 "github.com/openshift/client-go/config/applyconfigurations/config/v1"
+
+	configv1 "github.com/openshift/api/config/v1"
+
+	"github.com/davecgh/go-spew/spew"
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/test/e2e/framework"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+var _ = g.Describe("[sig-apimachinery]", func() {
+
+	defer g.GinkgoRecover()
+
+	oc := exutil.NewCLIWithPodSecurityLevel("server-side-apply-examples", admissionapi.LevelPrivileged)
+	fieldManager := metav1.ApplyOptions{
+		FieldManager: "e2e=test",
+	}
+
+	g.Describe("server-side-apply should function properly", func() {
+		g.It("should clear fields when they are no longer being applied", func() {
+			ctx := context.Background()
+
+			// TODO use another API.  I might have a version in my cache somewhere using a pod.
```

Comment:
Use a type where the conditions have `listType` and `listMapKey` annotations.

Reply:
> Use a type where the conditions have `listType` and `listMapKey` annotations.

Updated to have tests with
1. clusteroperator
2. featuregate
3. pods

behavior is consistent.  This could be cleaned up in to a decent e2e test if it's valuable.